Trackers/centroid_tracker/centroid.py

```diff
@@ -1,30 +1,112 @@
 import numpy as np
+from requests import delete
 from ..track import Track
+from scipy.spatial import distance as dist
 
 class centroid_track(Track):
-    def __init__(self, id, bbox, centroid):
+    def __init__(self, id, bbox, centroid, hits, miss):
         self.centroid = centroid
-        Track.__init__(self, id, bbox)
+        Track.__init__(self, id, bbox, hits, miss)
 
 class Centroid_tracker():
     def __init__(self):
         self.tracks = []
+        self.nextID=0
         self.max_age = 10
         self.min_hits = 3
         self.thres_distance = 20
 
     def add_track(self, id, bbox, centroid):
         """Create a centroid track object and adds to tracks list
         """
-        pass
+
+        '''
+        - add centroid to track list
+        - id++
+        - add instance from class track
+        - if hits > min_hits add track
+        '''
+
+        track = centroid_track(id, bbox, centroid, 1, 0)
+        self.tracks.append(track)
+        self.nextID +=1
+        print("added id ",track.id, "succesfully")
 
     def delete_track(self):
         """Deletes a centroid track object from tracks list if the max age is crossed.
         """
-        pass
+        for track in self.tracks:
+            if(track.miss > self.max_age):
+                print("deleted id ",track.id, "succesfully")
+                self.tracks.remove(track)
 
     def update(self, detections):
         """Returns the active list of centroid track objects"""
-        pass
+        
+        '''
+        - check if bbox is empty, if its empty, increase miss and check if miss> max_age if it is delete_track and return
+        - create centroid array, n calculate centroid of all tracks
+        - if nextid is 0, means no objects are tracking, then add_track
+        - else, find the min. distance of old centroid n new centroid
+            - check whether the centroid is matched to any row or col, if not then this means this object was lost earlier and now found, and reset it miss to 0
+            - check if input centroids is less than last time, so some objects are lost, and icrease their miss and check if it >max_age deregister it.
+                - if not, then we new objects, so register them
+        '''
+        if(detections==[] and self.tracks==[]):
+            return self.tracks
+        if(detections==[] and self.tracks!=[]):
```

Comment:
```suggestion
        if(detections==[]):
            if(self.tracks == []):
                 return self.tracks
            else:
```
we can use nested conditional statements here.